src/main/scala/com.snowplowanalytics/forex/Forex.scala

```diff
@@ -155,7 +160,7 @@ case class ForexLookupTo(conversionAmount: Double, fromCurr: String, fx: Forex)
  * @param toCurr   - The target currency
  * @param fx       - Forex object
  */
-case class ForexLookupWhen(conversionAmount: Double, fromCurr: String, toCurr: String, fx: Forex) {
+case class ForexLookupWhen[F[_]: Sync](conversionAmount: Double, fromCurr: String, toCurr: String, fx: Forex[F]) {
```

Comment:
Generally it is much more preferably to parametrize function by effects, not data/classes. This would allow us:

1. To construct entities in other functions without any constraints at all (e.g. just `ForexLookupWhen[F[_]]`
2. Finding other functions where effects can be much weaker (`Functor` or at least most of them are `ApplicationError[Throwable, ?]`, I believe.

It aligns nicely with tagless approach.

Reply:
E.g. I don't see in `getClient` anything that requires `Sync`.

Reply:
Re the second comment, OerClient requires a `Sync` implicit, so we have to take `Sync` as an implicit parameter also in `getClient`.

Reply:
Yes, but the whole point that `OerClient` *constructor* should not require it. It should be required by particular function, not by data type.